core/state_transition.go

```diff
@@ -236,8 +229,8 @@ func (st *StateTransition) buyGas() error {
 		balanceCheck = balanceCheck.Mul(balanceCheck, st.gasFeeCap)
 		balanceCheck.Add(balanceCheck, st.value)
 		if st.evm.ChainConfig().Scroll.FeeVaultEnabled() {
-			// always add l1fee, because all tx are L2-to-L1 ATM
-			balanceCheck.Add(balanceCheck, st.l1Fee)
+			// always add l1DataFee, because all tx are L2-to-L1 ATM
```

Comment:
ditto

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/351/commits/9e223e9db85603a82aac0a859c28ffa4045cbfeb